Test collection failing when building wheels

See https://github.com/mypyc/mypy_mypyc-wheels/actions, e.g. https://github.com/mypyc/mypy_mypyc-wheels/actions/runs/2695060465.

Test output:
```
____________ ERROR collecting site-packages/mypy/test/testcheck.py _____________
  site-packages/mypy/test/testcheck.py:42: in <module>
      typecheck_files.remove('check-python310.test')
  E   ValueError: list.remove(x): x not in list
  ____________ ERROR collecting site-packages/mypy/test/testcheck.py _____________
  site-packages/mypy/test/testcheck.py:42: in <module>
      typecheck_files.remove('check-python310.test')
  E   ValueError: list.remove(x): x not in list
  ____________ ERROR collecting site-packages/mypy/test/testcheck.py _____________
  site-packages/mypy/test/testcheck.py:42: in <module>
      typecheck_files.remove('check-python310.test')
  E   ValueError: list.remove(x): x not in list
  ____________ ERROR collecting site-packages/mypy/test/testparse.py _____________
  site-packages/mypy/test/testparse.py:15: in <module>
      class ParserSuite(DataSuite):
  site-packages/mypy/test/testparse.py:21: in ParserSuite
      files.remove('parse-python310.test')
  E   ValueError: list.remove(x): x not in list
  ____________ ERROR collecting site-packages/mypy/test/testparse.py _____________
  site-packages/mypy/test/testparse.py:15: in <module>
      class ParserSuite(DataSuite):
  site-packages/mypy/test/testparse.py:21: in ParserSuite
      files.remove('parse-python310.test')
  E   ValueError: list.remove(x): x not in list
  ____________ ERROR collecting site-packages/mypy/test/testparse.py _____________
  site-packages/mypy/test/testparse.py:15: in <module>
      class ParserSuite(DataSuite):
  site-packages/mypy/test/testparse.py:21: in ParserSuite
      files.remove('parse-python310.test')
  E   ValueError: list.remove(x): x not in list
  ___________ ERROR collecting site-packages/mypy/test/testsemanal.py ____________
  site-packages/mypy/test/testsemanal.py:37: in <module>
      semanal_files.remove('semanal-python310.test')
  E   ValueError: list.remove(x): x not in list
  ___________ ERROR collecting site-packages/mypy/test/testsemanal.py ____________
  site-packages/mypy/test/testsemanal.py:37: in <module>
      semanal_files.remove('semanal-python310.test')
  E   ValueError: list.remove(x): x not in list
  ___________ ERROR collecting site-packages/mypy/test/testsemanal.py ____________
  site-packages/mypy/test/testsemanal.py:37: in <module>
      semanal_files.remove('semanal-python310.test')
  E   ValueError: list.remove(x): x not in list
```

This was caused by #13143.

--- a/mypy/test/helpers.py
+++ b/mypy/test/helpers.py
@@ -22,7 +22,7 @@ from mypy.options import Options
 from mypy.test.data import (
     DataDrivenTestCase, fix_cobertura_filename, UpdateFile, DeleteFile
 )
-from mypy.test.config import test_temp_dir
+from mypy.test.config import test_temp_dir, test_data_prefix
 import mypy.version
 
 skip = pytest.mark.skip
@@ -500,6 +500,6 @@ def normalize_file_output(content: List[str], current_abs_path: str) -> List[str
 def find_test_files(pattern: str, exclude: Optional[List[str]] = None) -> List[str]:
     return [
         path.name
-        for path in (pathlib.Path("./test-data/unit").rglob(pattern))
+        for path in (pathlib.Path(test_data_prefix).rglob(pattern))
         if path.name not in (exclude or [])
     ]

